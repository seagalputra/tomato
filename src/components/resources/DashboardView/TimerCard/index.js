import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/common/Button'

const status = {
  FOCUS: 'Focus',
  LONG_BREAK: 'Long Break',
  SHORT_BREAK: 'Short Break'
}

const TimerCard = ({ onStart, onStop, state }) => {
  return (
    <div className="flex flex-col border rounded p-6 max-w-md">
      <div className="flex justify-center">
        <h1
          className={`${
            state === 'LONG_BREAK' || state === 'SHORT_BREAK'
              ? 'text-green-600'
              : 'text-red-600'
          } font-bold text-xl`}
        >
          {status[state]}
        </h1>
      </div>

      <div className="flex justify-center mt-2">
        <p className="text-gray-800 font-bold text-5xl">25:00</p>
      </div>

      <div className="inline-flex justify-center w-auto mt-4">
        <Button className="mr-2" onClick={onStart} variant="primary">
          Start
        </Button>

        <Button onClick={onStop} variant="secondary">
          Stop
        </Button>
      </div>
    </div>
  )
}

TimerCard.propTypes = {
  state: PropTypes.string.isRequired,
  onStart: PropTypes.func,
  onStop: PropTypes.func
}

export default TimerCard
