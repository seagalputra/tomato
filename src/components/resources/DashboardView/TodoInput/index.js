import React from 'react'

const TodoInput = () => {
  return (
    <form className="w-full p-2">
      <div className="block md:flex items-center">
        <input
          className="appearance-none w-full text-gray-700 mr-3 py-2 px-2 leading-light border border-teal-500 rounded"
          type="text"
          placeholder="To Do"
          aria-label="To Do"
        />
        <button
          className="flex-shrink-0 bg-teal-500 hover:bg-teal-700 border-teal-500 hover:border-teal-700 text-sm border-4 text-white py-2 px-2 rounded w-full md:w-32 mt-3 md:mt-0"
          type="button"
        >
          Add
        </button>
      </div>
    </form>
  )
}

export default TodoInput
