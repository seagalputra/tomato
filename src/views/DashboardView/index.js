import React, { useState, useEffect } from 'react'

import TimerCard from 'components/resources/DashboardView/TimerCard'

import timerImage from 'assets/img/timer.svg'

const getTimer = state => {
  if (state === 'SHORT_BREAK') return 300

  if (state === 'LONG_BREAK') return 900

  return 1500
}

const changeTimerState = currentIteration => {
  if (currentIteration <= 6) {
    if (currentIteration % 2 === 0) return 'FOCUS'

    return 'SHORT_BREAK'
  }

  return 'LONG_BREAK'
}

const DashboardView = () => {
  const [timer, setTimer] = useState(1500)
  const [timerState, setTimerState] = useState('FOCUS')
  const [iteration, setIteration] = useState(0)
  const [isActive, setIsActive] = useState(false)

  const onStartTimer = () => {
    setIsActive(!isActive)
    setIteration(iteration + 1)
  }

  const onStopTimer = state => {
    setTimer(getTimer(state))
    setIsActive(false)
  }

  useEffect(() => {
    let interval = null

    if (isActive) {
      interval = setInterval(() => {
        setTimer(time => time - 1)
      }, 1000)

      if (timer === 0) {
        clearInterval(interval)
        setTimerState(changeTimerState(iteration))
        setIsActive(false)
      }
    } else if (!isActive && timer !== 0) {
      clearInterval(interval)
    } else {
      setTimer(getTimer(timerState))
    }

    return () => clearInterval(interval)
  }, [isActive, timer, timerState, iteration])

  return (
    <main>
      <TimerCard
        timer={timer}
        state={timerState}
        onStart={onStartTimer}
        onStop={onStopTimer}
        isActive={isActive}
      />

      <div className="flex flex-col px-6 py-16 border-b">
        <img
          className="w-56 self-center"
          src={timerImage}
          alt="Timer illustration"
        />
        <p className="text-2xl text-center font-bold text-gray-800 mt-4">
          What is Pomodoro?
        </p>
        <div className="mt-3 w-40 self-center border-b-2 border-gray-400" />
        <p className="text-base text-center text-gray-800 mt-6 px-16">
          Pomodoro technique is one of the time management technique. This
          method developed by Francesco Cirillo in late 1980s. This technique
          uses a timer to break down work into intevals, commonly use 25 minutes
          in length, separated by short breaks. Each interval is known as a
          pomodoro, from the italian word from &apos;tomato&apos;, after the
          tomato-shaped kitchen timer that Cirillo used as a university student.
        </p>
      </div>
    </main>
  )
}

export default DashboardView
