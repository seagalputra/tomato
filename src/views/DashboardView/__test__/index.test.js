import React from 'react'
import { shallow } from 'enzyme'
import DashboardView from '..'

describe('DashboardView', () => {
  it('renders without crashing', () => {
    const component = shallow(<DashboardView />)
    expect(component).toMatchSnapshot()
  })
})
