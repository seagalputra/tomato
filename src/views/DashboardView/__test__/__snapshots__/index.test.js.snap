// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardView renders without crashing 1`] = `
<Fragment>
  <TimerCard
    isActive={false}
    onStart={[Function]}
    onStop={[Function]}
    state="FOCUS"
    timer={1500}
  />
  <div
    className="flex flex-col px-6 py-16"
  >
    <img
      alt="Timer illustration"
      className="w-56 self-center"
      src="timer.svg"
    />
    <p
      className="text-2xl text-center font-bold text-gray-800 mt-4"
    >
      What is Pomodoro?
    </p>
    <div
      className="mt-3 w-40 self-center border-b-2 border-gray-400"
    />
    <p
      className="text-base text-center text-gray-800 mt-6 px-16"
    >
      Pomodoro technique is one of the time management technique. This method developed by Francesco Cirillo in late 1980s. This technique uses a timer to break down work into intevals, commonly use 25 minutes in length, separated by short breaks. Each interval is known as a pomodoro, from the italian word from 'tomato', after the tomato-shaped kitchen timer that Cirillo used as a university student.
    </p>
  </div>
</Fragment>
`;
